lib/tether/config.go

```diff
@@ -88,3 +89,32 @@ type SessionConfig struct {
 	Errwriter dio.DynamicMultiWriter
 	Reader    dio.DynamicMultiReader
 }
+
+type NetworkEndpoint struct {
+	// Common.Name - the nic alias requested (only one name and one alias possible in linux)
+	// Common.ID - pci slot of the vnic allowing for interface identifcation in-guest
+	metadata.Common
+
+	// IP address to assign - nil if DHCP
+	Static *net.IPNet `vic:"0.1" scope:"read-only" key:"staticip"`
+
+	// Actual IP address assigned
+	Assigned net.IPNet `vic:"0.1" scope:"read-write" key:"ip"`
+
+	// The network in which this information should be interpreted. This is embedded directly rather than
+	// as a pointer so that we can ensure the data is consistent
+	Network metadata.ContainerNetwork `vic:"0.1" scope:"read-only" key:"network"`
+
+	// DHCP runtime info
+	DHCP *DHCPInfo `vic:"0.1" scope:"read-only" recurse:"depth=0"`
```

Comment:
Actually I don't want to serialize it at all. Without the annotation I was getting was not DHCP == nil, which I need in the code.